### ISSUE TITLE
[4.0.0-beta.16] ion-slides not working properly on fresh page load

### ISSUE THREAD

**Issue by bergben** [ionic-team/ionic#16364]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```Ionic:

   ionic (Ionic CLI)             : 4.3.1 (C:\Users\***\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework               : @ionic/angular 4.0.0-beta.16
   @angular-devkit/build-angular : 0.10.6
   @angular-devkit/schematics    : 7.0.6
   @angular/cli                  : 7.0.6
   @ionic/angular-toolkit        : 1.2.0

System:

   NodeJS : v10.13.0 (C:\Program Files\nodejs\node.exe)
   npm    : 5.6.0
   OS     : Windows 10
```

**Describe the Bug**
Ionic Slides Component is not working when the page is refreshed.

**Steps to Reproduce**
1. ionic start tester tabs --type=angular
2. In home.html add Slides like so:
```<ion-content padding>
    <ion-slides>
        <ion-slide>
            <h1>Slide 1</h1>
        </ion-slide>
        <ion-slide>
            <h1>Slide 2</h1>
        </ion-slide>
        <ion-slide>
            <h1>Slide 3</h1>
        </ion-slide>
    </ion-slides>
</ion-content>
```
3. ionic serve
4. Refresh the page where the slides are on with F5 (without needing to navigate)
5. Slides component is broken (sliding not working etc.)

**Related Code**
See steps to reproduce

**Expected Behavior**
Slides component should still work as expected.

**Comment by bergben** [comment]
Still an issue in [4.0.0-beta.17], makes the Slides unusable in its current state...

**Comment by bryce13950** [comment]
Could you describe a bit what is happening? Is it just completely broken? Is the animation messed up?

**Comment by bergben** [comment]
I think it's best to show with a screen capture:
Minimal example with starter app: https://imgur.com/a/vh8WK67
Actual app with the problem: https://imgur.com/a/1CpG38c 

I noticed it does not happen 100% of the time, but a good 50% at least so if you just follow the steps described I think you should be able to reproduce the issue. 

If you need any other information please let me know!

**Comment by bryce13950** [comment]
Ahhh is it that it is not snapping into place?

**Comment by bergben** [comment]
Basically yes. The content is not loading correctly either, but I am quite sure that is fixed once the snapping is fixed 👍

**Comment by bryce13950** [comment]
Well if you could provide a repository with the source code already ready that would be super helpful. If you can't then that is fine, but it would make fixing the issue itself much easier!

**Comment by bergben** [comment]
Sure, there you go: https://github.com/bergben/ionic_issue_16364_slides
Just clone the repo, run npm i and ionic serve. I included a description on possible methods to reproduce the issue in the home tab! 

Let me know if you need anything else!

**Comment by bryce13950** [comment]
Thanks!

**Comment by bergben** [comment]
I figured out what the problem is: The slides component won't work if it is wrapped in a div that applies [hidden] or *ngIf or if <ion-slides> applies it itself.

**Comment by liamdebeasi** [comment]
A fix for this is available in Ionic 4.9.0. For any new bugs, please create a new issue.

Thanks!
